service/src/java/org/apache/hive/service/cli/thrift/RetryingThriftCLIServiceClient.java

```diff
@@ -310,7 +310,7 @@ protected synchronized TTransport connect(HiveConf conf) throws HiveSQLException
 
     String host = conf.getVar(HiveConf.ConfVars.HIVE_SERVER2_THRIFT_BIND_HOST);
     int port = conf.getIntVar(HiveConf.ConfVars.HIVE_SERVER2_THRIFT_PORT);
-    int maxThriftMessageSize = (int) conf.getSizeVar(HiveConf.ConfVars.HIVE_THRIFT_CLIENT_MAX_MESSAGE_SIZE);
+    int maxThriftMessageSize = (int) Math.min(conf.getSizeVar(HiveConf.ConfVars.HIVE_THRIFT_CLIENT_MAX_MESSAGE_SIZE),Integer.MAX_VALUE);
     LOG.info("Connecting to " + host + ":" + port);
```

Comment:
It might be a good idea to also include the maxThriftMessageSize in the below LOG.info message. It would help in future debugging if needed (and that LOG.info line already exists, so we likely won't be adding a very chatty message).

Reply:
I think it should be ok to change L#314 as follows:
LOG.info("Connecting to " + host + ":" + port+ " using a thrift max message of size: "+ maxThriftMessageSize);

Reply:
Thanks for reviewing @jfsii and @saihemanth-cloudera! Will incorporate this.